storages/auth/s3_v2.go

```diff
@@ -250,15 +251,50 @@ type signRoundTripper struct {
 	keys Keys
 }
 
+type signAuthServiceRoundTripper struct {
+	rt      http.RoundTripper
+	crd     *crdstore.CredentialsStore
+	backend string
+}
+
 // RoundTrip implements http.RoundTripper interface
 func (srt signRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
 	SignRequestV2(req, srt.keys)
 	return srt.rt.RoundTrip(req)
 }
 
+// RoundTrip implements http.RoundTripper interface
+func (srt signAuthServiceRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	accessKey := strings.Trim(req.Header.Get("Authorization"), " ")
+	start := strings.IndexAny(accessKey, " ")
+	end := strings.IndexAny(accessKey, ":")
+	if start == -1 || end == -1 {
+		return &http.Response{StatusCode: http.StatusBadRequest, Request: req}, fmt.Errorf("cannot find AWS AccessKey in request")
+	}
+	accessKey = accessKey[start+1 : end]
```

Comment:
Extract code above to separate pure function that will extract access key from a header.

Reply:
you can use `strings.SplitN` for extracting values from header `Authorization` - example: [https://gist.github.com/sambengtson/bc9f76331065f09e953f]